homeassistant/components/sonos/speaker.py

```diff
@@ -802,26 +802,56 @@ async def restore_multi(
         """Restore snapshots for all the speakers."""
 
         def _restore_groups(
-            speakers: list[SonosSpeaker], with_group: bool
+            speakers: set[SonosSpeaker], with_group: bool
         ) -> list[list[SonosSpeaker]]:
             """Pause all current coordinators and restore groups."""
             for speaker in (s for s in speakers if s.is_coordinator):
-                if speaker.media.playback_status == SONOS_STATE_PLAYING:
-                    speaker.soco.pause()
+                if (
+                    speaker.media.playback_status == SONOS_STATE_PLAYING
+                    and "Pause" in speaker.soco.available_actions
+                ):
+                    try:
+                        speaker.soco.pause()
+                    except SoCoUPnPException as exc:
+                        _LOGGER.debug(
+                            "Pause failed during restore of %s: %s",
+                            speaker.zone_name,
+                            speaker.soco.available_actions,
+                            exc_info=exc,
+                        )
 
             groups = []
```

Comment:
Could save some indent and return early

```suggestion
            groups = []
            if not with_group:
               return groups
```

then outdent below